addons/account/models/account_bank_statement.py

```diff
@@ -182,6 +182,36 @@ def _compute_is_complete(self):
             stmt.is_complete = stmt.line_ids and stmt.currency_id.compare_amounts(
                 stmt.balance_end, stmt.balance_end_real) == 0
 
+        # Process the bank statement report.
+        if self.ids:
+            complete_stmt = self.filtered('is_complete')
```

Comment:
I have an odd feeling that it is going to generate the reports for not only the statements that have just been completed, but also the already completed ones.
Maybe this?
```js
    @api.depends('balance_end_real', 'balance_end')
    def _compute_is_complete(self):
        incomplete_statements = self.filtered(lambda stmt: not stmt.is_complete)
        for stmt in self:
            "..."
        complete_stmt = self.filtered('is_complete')
        completed_statements = incomplete_statements & complete_stmt
        if completed_statements.ids:
            "..."
 ```